server/model/monitor.js

```diff
@@ -457,12 +459,28 @@ class Monitor extends BeanModel {
         if (this.type === "push") {
             setTimeout(() => {
                 safeBeat();
-            }, this.interval * 1000);
+            }, this.getNextInterval() * 1000);
         } else {
             safeBeat();
         }
     }
 
+    getNextInterval() {
+        if (!this.cron || /^\s*$/.test(this.cron)) {
+            return this.interval;
+        }
+        const laterCron = later.parse.cron(this.cron);
+        const nextOccurences = later.schedule(laterCron).next(2);
+
+        let calculatedInterval = Math.floor((nextOccurences[0].getTime() - Date.now()) / 1000);
+        if (calculatedInterval < 20) { //beat interval too low,  using next interval
```

Comment:
Maybe in previous line (or next)?
```suggestion
        if (calculatedInterval < 20) { // beat interval too low,  using next interval
```